quantlib/pricingengines/vanilla/vanilla.pyx

```diff
@@ -52,6 +62,72 @@ cdef class AnalyticHestonEngine(PricingEngine):
             )
         )
 
+cdef class AnalyticBSMHullWhiteEngine(PricingEngine):
+
+    def __init__(self, Real equity_short_rate_correlation,
+            GeneralizedBlackScholesProcess process,
+            HullWhite hw_model):
+
+        cdef shared_ptr[_bsp.GeneralizedBlackScholesProcess] process_ptr = \
+            shared_ptr[_bsp.GeneralizedBlackScholesProcess](
+                deref(process._thisptr)
+            )
+
+        self._thisptr = new shared_ptr[_vanilla.PricingEngine](
+            new _vanilla.AnalyticBSMHullWhiteEngine(
+                equity_short_rate_correlation,
+                process_ptr,
+                deref(<shared_ptr[_hw.HullWhite]*> hw_model._thisptr)
+            )
+        )
+    
+
+cdef class AnalyticHestonHullWhiteEngine(PricingEngine):
+
+    def __init__(self, HestonModel heston_model,
+                 HullWhite hw_model,
+                 int integration_order=144):
+
+        self._thisptr = new shared_ptr[_vanilla.PricingEngine](
+            new _vanilla.AnalyticHestonHullWhiteEngine(
+                deref(heston_model._thisptr),
+                deref(<shared_ptr[_hw.HullWhite]*> hw_model._thisptr),
+                <Size>integration_order
+            )
+        )
+
+
+cdef class FdHestonHullWhiteVanillaEngine(PricingEngine):
+
+    def __init__(self, HestonModel heston_model,
+            HullWhiteProcess hw_process,
+            Real corrEquityShortRate,
+            Size tGrid,
+            Size xGrid, 
+            Size vGrid,
+            Size rGrid,
+            Size dampingSteps,
+            bool controlVariate,
+            FdmSchemeDesc desc):
+
+        self._thisptr = new shared_ptr[_vanilla.PricingEngine](
+            new _vanilla.FdHestonHullWhiteVanillaEngine(
+                deref(heston_model._thisptr),
+                deref(hw_process._thisptr),
+                corrEquityShortRate,
+                tGrid,
+                xGrid, 
+                vGrid,
+                rGrid,
+                dampingSteps,
+                controlVariate,
+                deref(desc._thisptr.get()))
+            )
+
+    def enableMultipleStrikesCaching(self, strikes):
```

Comment:
We should check the conventions for method naming and be consistent at the library level. Ideally, this would be `enable_multiple_strikes_catching` and not `enableMultipleStrikesCaching`